catalog/provider.go

```diff
@@ -391,7 +392,7 @@ func (prov *DatabaseProvider) HasDatabase(ctx *sql.Context, name string) bool {
 	prov.mu.RLock()
 	defer prov.mu.RUnlock()
 
-	ok, err := hasDatabase(ctx, prov.catalogName, name)
+	ok, err := hasDatabase(ctx, prov.pool.CurrentCatalog(ctx.ID()), name)
```

Comment:
Maybe we could add a `CurrentCatalog` wrapper to the `adapter` package to hide the implementation `pool.CurrentCatalog(ctx.ID())`. In this way, we can get the underlying catalog without explicit access to the `pool`.